backend/baton/src/test/java/touch/baton/domain/runnerpost/command/service/RunnerPostCommandServiceEventTest.java

```diff
@@ -0,0 +1,119 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.event.ApplicationEvents;
+import org.springframework.test.context.event.RecordApplicationEvents;
+import touch.baton.config.ServiceTestConfig;
+import touch.baton.domain.member.command.Member;
+import touch.baton.domain.member.command.Runner;
+import touch.baton.domain.member.command.Supporter;
+import touch.baton.domain.runnerpost.command.RunnerPost;
+import touch.baton.domain.runnerpost.command.event.RunnerPostApplySupporterEvent;
+import touch.baton.domain.runnerpost.command.event.RunnerPostAssignSupporterEvent;
+import touch.baton.domain.runnerpost.command.event.RunnerPostReviewStatusDoneEvent;
+import touch.baton.domain.runnerpost.command.service.dto.RunnerPostApplicantCreateRequest;
+import touch.baton.domain.runnerpost.command.service.dto.RunnerPostUpdateRequest;
+import touch.baton.fixture.domain.MemberFixture;
+import touch.baton.fixture.domain.RunnerFixture;
+import touch.baton.fixture.domain.RunnerPostFixture;
+import touch.baton.fixture.domain.SupporterFixture;
+
+import static java.time.LocalDateTime.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static touch.baton.fixture.vo.DeadlineFixture.deadline;
+
+@RecordApplicationEvents
+class RunnerPostCommandServiceEventTest extends ServiceTestConfig {
+
+    @Autowired
+    private ApplicationEvents applicationEvents;
+
+    private RunnerPostCommandService runnerPostCommandService;
+
+    @BeforeEach
+    void setUp() {
+        runnerPostCommandService = new RunnerPostCommandService(
+                runnerPostCommandRepository,
+                tagCommandRepository,
+                supporterCommandRepository,
+                supporterRunnerPostCommandRepository,
+                publisher
+        );
+    }
+
+    @DisplayName("서포터가 러너 게시글에 리뷰를 지원하면 이벤트가 발행된다.")
+    @Test
+    void success_supporter_apply_runnerPost() {
+        // given
+        final Member savedMemberDitoo = memberCommandRepository.save(MemberFixture.createDitoo());
+        final Runner savedRunnerDitto = runnerQueryRepository.save(RunnerFixture.createRunner(savedMemberDitoo));
+
+        final Member savedMemberHyena = memberCommandRepository.save(MemberFixture.createHyena());
+        final Supporter savedSupporterHyena = supporterCommandRepository.save(SupporterFixture.create(savedMemberHyena));
+
+        final RunnerPost savedRunnerPost = runnerPostCommandRepository.save(RunnerPostFixture.create(savedRunnerDitto, deadline(now().plusHours(100))));
+
+        // when
+        final RunnerPostApplicantCreateRequest request = new RunnerPostApplicantCreateRequest("안녕하세요. 서포터 헤나입니다.");
+        runnerPostCommandService.createRunnerPostApplicant(savedSupporterHyena, request, savedRunnerPost.getId());
+
+        // then
+        final int eventPublishedCount = (int) applicationEvents.stream(RunnerPostApplySupporterEvent.class).count();
```

Comment:
```suggestion
        final int eventPublishedCount = Long.valueOf(applicationEvents.stream(RunnerPostApplySupporterEvent.class).count()).intValue();
```
타입캐스팅보다 이렇게 하거나, 그냥 Long 값으로 받는 건 어때요? 타입캐스팅하면 int 범위 벗어나도 예외가 안터져서요...

Reply:
```suggestion
        final int eventPublishedCount = applicationEvents.stream(RunnerPostApplySupporterEvent.class).mapToInt(e -> 1).sum();
```
javadoc에 나온것처럼 하면 이런 방법도 있음
근데 long으로 그냥 쓰는게 더 좋을거 가탕요

Reply:
테스트 코드라서 long으로 그냥 쓰는게 좋을거 같다에 한 표입니다!